Makefile

```
@@ -35,8 +35,8 @@ endif
 DUCKDB_LIB = libduckdb$(DLSUFFIX)
 FULL_DUCKDB_LIB = third_party/duckdb/build/$(DUCKDB_BUILD_TYPE)/src/$(DUCKDB_LIB)
 
-override PG_CPPFLAGS += -Iinclude -Ithird_party/duckdb/src/include -Ithird_party/duckdb/third_party/re2
-override PG_CXXFLAGS += -std=c++17 -Wno-sign-compare -Wno-register ${DUCKDB_BUILD_CXX_FLAGS}
+override PG_CPPFLAGS += -Iinclude -isystem third_party/duckdb/src/include -isystem third_party/duckdb/third_party/re2
+override PG_CXXFLAGS += -std=c++17 -Wno-sign-compare -Wno-register -Wshadow -Wswitch -Wunused-parameter -Wunreachable-code -Wno-unknown-pragmas ${DUCKDB_BUILD_CXX_FLAGS} -Wall -Wextra -Werror
```

```suggestion
override PG_CXXFLAGS += -std=c++17 -Wno-sign-compare -Wno-register -Wshadow -Wswitch -Wunused-parameter -Wunreachable-code -Wno-unknown-pragmas ${DUCKDB_BUILD_CXX_FLAGS} -Wall -Wextra
```

`-Werror` should be opt-in (and turned on in CI). It's way to common for different compilers to have some extra warnings, and there's no need to fail compilation for someone building the pg_duckdb on a different system than CI is using.